### PR TITLE
Contact Form : Validations and sending email

### PR DESCRIPTION
**Pull request by meriouma**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,20 +22,23 @@
...
+        <gwtp.version>1.4</gwtp.version>
         <guice.version>3.0</guice.version>
         <gin.version>2.1.2</gin.version>
-        <gsss.version>1.0-SNAPSHOT</gsss.version>
-        <gwtquery.version>1.4.3-SNAPSHOT</gwtquery.version>
+        <gsss.version>1.0</gsss.version>
+        <gwtquery.version>1.4.3</gwtquery.version>
         <gwt-maps-api.version>3.10.0-alpha-7</gwt-maps-api.version>
         <tooltip.version>1.1</tooltip.version>
         <appear.version>1.0-SNAPSHOT</appear.version>
```
Comment (christiangoudreau) [inline pom.xml]: Still snapshot?

File: pom.xml [hunk 2]
```diff
@@ -22,20 +22,23 @@
...
         <gwtchosen.version>2.0.0-SNAPSHOT</gwtchosen.version>
```
Comment (christiangoudreau) [inline pom.xml]: Idem?

File: pom.xml [hunk 3]
```diff
@@ -22,20 +22,23 @@
...
+        <gwt-seo.version>0.1-SNAPSHOT</gwt-seo.version>
```
Comment (christiangoudreau) [inline pom.xml]: When do we do the official release? :D

File: pom.xml [hunk 4]
```diff
@@ -22,20 +22,23 @@
...
         <gwt.version>2.7.0</gwt.version>
-        <gae.version>1.9.9</gae.version>
-        <gwtp.version>1.5-SNAPSHOT</gwtp.version>
+        <gae.version>1.9.19</gae.version>
+        <gwtp.version>1.4</gwtp.version>
         <guice.version>3.0</guice.version>
         <gin.version>2.1.2</gin.version>
-        <gsss.version>1.0-SNAPSHOT</gsss.version>
-        <gwtquery.version>1.4.3-SNAPSHOT</gwtquery.version>
+        <gsss.version>1.0</gsss.version>
```
Comment (jasonlemay) [inline pom.xml]: You can use 1.1-SNAPSHOT

File: src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml [hunk 5]
```diff
@@ -12,22 +12,22 @@
...
+                <div>
                     <label>Name</label>
-                    <input type="text" name="name" autofocus="autofocus"/>
+                    <input ui:field="name" type="text" name="name" autofocus="autofocus"/>
                 </div>
-                <div class="{form.style.error}">
+                <div>
                     <label>Email</label>
-                    <input type="email" name="Email" required=""/>
+                    <input ui:field="email" name="Email"/>
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml]: We can validate the field with html5 without our JS logic @jasonlemay

File: src/main/java/com/arcbees/website/shared/ContactRequest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.arcbees.website.shared;
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/website/shared/ContactRequest.java]: license header

File: src/main/java/com/arcbees/website/server/SupportResource.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Inject
+    SupportResource(EmailSender emailSender) {
+        this.emailSender = emailSender;
+    }
+
+    @POST
+    public Response sendContactEmail(ContactRequest contactRequest) {
+        com.arcbees.appengine.mail.Email convertedEmail = EmailBuilder.to(TO)
+                .fromAddress(FROM)
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/website/server/SupportResource.java]: Isn't there any reply to?

File: src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml (hunk 5 above)
Comment (jasonlemay) [inline src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml]: @christiangoudreau yes, but not in this example, as the browser's default validation style was overwriting @LarryMatte styling. It's not a big problem to use JS validation like so as the app needs JS to run anyway.

Comment (jasonlemay) [inline src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml]: @meriouma `type="email"` would be required ... does it trigger browser validation?

Comment (meriouma) [inline src/main/java/com/arcbees/website/client/application/contactform/ContactFormView.ui.xml]: Yeah, it triggers the browser validation

File: src/main/java/com/arcbees/website/server/SupportResource.java (hunk 7 above)
Comment (meriouma) [inline src/main/java/com/arcbees/website/server/SupportResource.java]: Good idea, this needs to be added to appengine-mail

File: pom.xml (hunk 3 above)
Comment (meriouma) [inline pom.xml]: Trying to get the text done.. :/

**Pull request closed without merging**